extractor/src/main/java/org/schabi/newpipe/extractor/channel/ChannelTabInfo.java

```diff
@@ -0,0 +1,62 @@
+package org.schabi.newpipe.extractor.channel;
+
+import org.schabi.newpipe.extractor.InfoItem;
+import org.schabi.newpipe.extractor.ListExtractor;
+import org.schabi.newpipe.extractor.ListInfo;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.utils.ExtractorHelper;
+
+import java.io.IOException;
+import java.util.List;
+
+public class ChannelTabInfo extends ListInfo<InfoItem> {
+    public ChannelTabInfo(int serviceId, String id, String url, String originalUrl, String name, ListLinkHandler listLinkHandler) {
+        super(serviceId, id, url, originalUrl, name, listLinkHandler.getContentFilters(), listLinkHandler.getSortFilter());
+    }
+
+    public static ListExtractor.InfoItemsPage<InfoItem> getMoreItems(ChannelTabInfo tabInfo, String pageUrl) throws IOException, ExtractionException {
+        if (tabInfo.getChannelTabExtractor() == null) {
+            ChannelExtractor channelExtractor = NewPipe.getServiceByUrl(tabInfo.getUrl()).getChannelExtractor(tabInfo.getUrl());
+            channelExtractor.fetchPage();
+
+            List<ChannelTabExtractor> channelTabExtractors = channelExtractor.getTabs();
+            for (ChannelTabExtractor channelTabExtractor : channelTabExtractors) {
+                if (channelTabExtractor.getName().equals(tabInfo.getName())) {
+                    tabInfo.setChannelTabExtractor(channelTabExtractor);
+                }
+            }
+        }
+
+        return tabInfo.getChannelTabExtractor().getPage(pageUrl);
+    }
+
+    public static ChannelTabInfo getInfo(ChannelTabExtractor extractor) throws ExtractionException {
+        final int serviceId = extractor.getServiceId();
+        final String id = extractor.getId();
+        final String url = extractor.getUrl();
+        final String originalUrl = extractor.getOriginalUrl();
+        final String name = extractor.getName();
+
+        final ChannelTabInfo info = new ChannelTabInfo(serviceId, id, url, originalUrl, name, extractor.getLinkHandler());
+
+        info.setChannelTabExtractor(extractor);
+
+        final ListExtractor.InfoItemsPage<InfoItem> itemsPage = ExtractorHelper.getItemsPageOrLogError(info, extractor);
+        info.setRelatedItems(itemsPage.getItems());
+        info.setNextPageUrl(itemsPage.getNextPageUrl());
+
+        return info;
+    }
+
+    private transient ChannelTabExtractor channelTabExtractor;
+
+    public ChannelTabExtractor getChannelTabExtractor() {
+        return channelTabExtractor;
+    }
+
+    public void setChannelTabExtractor(ChannelTabExtractor channelTabExtractor) {
+        this.channelTabExtractor = channelTabExtractor;
+    }
```

Comment:
I thought that `Info` objects were supposed to be plain java/data objects? Storing a extractor in it feels definitely wrong (not serializable as well).

Would like your opinion/suggestions on this.

Reply:
True, that's why the `channelTabExtractor` variable is `transient`. This is needed to be able to get more items. The `getMoreItems()` function sets the `channelTabExtractor` again if it's not there anymore. We're doing this in `CommentsInfo` as well.

Reply:
> True, that's why the channelTabExtractor variable is transient

I noticed it, I was talking about the use of the extractor object at all in the data class.

> This is needed to be able to get more items.

Is it though? Aren't we able to just use the [nextPageUrl](url) to get the next page? We are able to that in `ChannelInfo` itself currently.

But now we don't even need to fetch the initial page because the continuation response already include all we need.